library/core/src/ptr/mut_ptr.rs

```diff
@@ -817,6 +819,12 @@ impl<T: ?Sized> *mut T {
     /// (Note that [`offset`] and [`add`] also have a similar limitation and hence cannot be used on
     /// such large allocations either.)
     ///
+    /// The requirement for pointers to be derived from the same allocated object is primarily
+    /// needed for `const`-compatibility: at compile-time, pointers into *different* allocated
+    /// object do not have a known distance to each other. However, the requirement also exists at
+    /// runtime, and may be exploited by optimizations. You can use `(self as usize).sub(origin as
```

Comment:
```suggestion
    /// runtime and may be exploited by optimizations. You can use `(self as usize).sub(origin as
```

The reason this doesn't get to have a comma is the second part isn't a proper sentence on its own and this isn't a 3+ item list.